crates/sqlexec/src/remote/client.rs

```diff
@@ -199,35 +204,44 @@ impl RemoteClient {
         metadata.insert(DB_NAME_KEY, params.db_name.parse()?);
         metadata.insert(ORG_KEY, params.org.parse()?);
 
+        let mut body = HashMap::new();
+        body.insert("user", params.user);
+        body.insert("password", params.password);
+        body.insert("org_name", params.org);
+        body.insert("db_name", params.db_name);
+        body.insert("api_version", 2.to_string());
+        body.insert("client_type", client_type.to_string());
+
+        info!("client authentication");
+        let http_client = reqwest::Client::new();
+        let res = http_client
+            .post(format!(
+                "{}/api/internal/authenticate/client",
+                cloud_api_addr
+            ))
+            .json(&body)
+            .send()
+            .await?;
+
+        if res.status() != reqwest::StatusCode::OK {
+            if res.status().is_client_error() {
+                let err = res.json::<AuthenticateClientError>().await?;
+                return Err(ExecError::String(err.msg));
+            } else {
+                return Err(ExecError::Internal("client authentication".to_string()));
+            }
+        }
+
         let mut dst: Endpoint = dst.try_into()?;
 
         if !disable_tls {
-            info!("apply TLS certificate to rpc proxy connection");
-
-            let mut body = HashMap::new();
-            body.insert("user", params.user);
-            body.insert("password", params.password);
-            body.insert("org_name", params.org);
-            body.insert("db_name", params.db_name);
-            body.insert("api_version", 2.to_string());
-            body.insert("client_type", client_type.to_string());
-
-            let client = reqwest::Client::new();
-            let res = client
-                .post(format!(
-                    "{}/api/internal/authenticate/client",
-                    cloud_api_addr
-                ))
-                .json(&body)
-                .send()
-                .await?
-                .json::<AuthenticateClientResponse>()
-                .await?;
+            info!("apply TLS certificate");
```

Comment:
debug

Reply:
This was there before, and I don't mind keeping it in for when verbose logs are on.